src/Connector.sol

```diff
@@ -111,7 +122,7 @@ contract CentrifugeConnector is Test {
 
```

Comment:
I wonder whether we actually need the check on line 116-117. It adds an `SLOAD` for the pool storage. But afaik through the check in `addTranche`, the value `tranches[poolId][trancheId]` cannot exist if the `poolId` does not already exist.

Reply:
Not sure I follow... Unless it was changed, the check I believe you're talking about is this one:
```
Pool storage pool = pools[poolId];
require(pool.createdAt > 0, "CentrifugeConnector/invalid-pool");
```

Just checking that the pool exists, not the tranche.

Reply:
@AStox In `deployTranche`, we are checking first if the pool exists, and then if the tranche exists. I think we don't need both checks, if the tranche exists then the pool also must exist.

Reply:
Actually it looks like we're only checking if the pool exists. That should probably be just checking if the tranche exists instead.